dart/lib/src/open_location_code.dart

```diff
@@ -1,598 +1,605 @@
+const separator = '+'; // 43 Ascii
+
+/// The number of characters to place before the separator.
+const separatorPosition = 8;
 
-// The number of characters to place before the separator.
-const int SEPARATOR_POSITION = 8;
+/// The character used to pad codes.
+const padding = '0'; // 48 in Ascii
 
-// The character used to pad codes.
-const String PADDING = '0'; // 48 in Ascii
+/// The character set used to encode the values.
+const codeAlphabet = '23456789CFGHJMPQRVWX';
 
-// The character set used to encode the values.
-const String CODE_ALPHABET = '23456789CFGHJMPQRVWX';
+/// The base to use to convert numbers to/from.
+const encodingBase = codeAlphabet.length;
 
-// The base to use to convert numbers to/from.
-const int ENCODING_BASE = CODE_ALPHABET.length;
+/// The maximum value for latitude in degrees.
+const latitudeMax = 90;
 
-// The maximum value for latitude in degrees.
-const int LATITUDE_MAX = 90;
+/// The maximum value for longitude in degrees.
+const longitudeMax = 180;
 
-// The maximum value for longitude in degrees.
-const int LONGITUDE_MAX = 180;
+/// Maximum code length using lat/lng pair encoding. The area of such a
+/// code is approximately 13x13 meters (at the equator), and should be suitable
+/// for identifying buildings. This excludes prefix and separator characters.
+const pairCodeLength = 10;
 
-// Maximum code length using lat/lng pair encoding. The area of such a
-// code is approximately 13x13 meters (at the equator), and should be suitable
-// for identifying buildings. This excludes prefix and separator characters.
-const int PAIR_CODE_LENGTH = 10;
+/// The resolution values in degrees for each position in the lat/lng pair
+/// encoding. These give the place value of each position, and therefore the
+/// dimensions of the resulting area.
+const pairResolutions = const <double>[20.0, 1.0, .05, .0025, .000125];
 
-// The resolution values in degrees for each position in the lat/lng pair
-// encoding. These give the place value of each position, and therefore the
-// dimensions of the resulting area.
-List<double> PAIR_RESOLUTIONS = [20.0, 1.0, .05, .0025, .000125];
+/// Number of columns in the grid refinement method.
+const gridColumns = 4;
 
-// Number of columns in the grid refinement method.
-const int GRID_COLUMNS = 4;
+/// Number of rows in the grid refinement method.
+const gridRows = 5;
 
-// Number of rows in the grid refinement method.
-const int GRID_ROWS = 5;
+/// Size of the initial grid in degrees.
+const gridSizeDegrees = 0.000125;
 
-// Size of the initial grid in degrees.
-const double GRID_SIZE_DEGREES = 0.000125;
+/// Minimum length of a code that can be shortened.
+const minTrimmableCodeLen = 6;
 
-// Minimum length of a code that can be shortened.
-const int MIN_TRIMMABLE_CODE_LEN = 6;
+/// Decoder lookup table.
+///
+/// * -2: illegal.
+/// * -1: Padding or Separator
+/// * >= 0: index in the alphabet.
+const _decode = const <int>[
```

Comment:
This appears to be a lookup table based on ASCII codes to whether a character is valid or not. It's tightly coupled to the codeAlphabet string above and I don't know why this table exists. If either the table or the codeAlphabet string are changed bad things happen.

Can we remove this and use codeAlphabet lookups to determine whether a character is valid and what it's value is? (Like all the other implementations, basically.)